### PR TITLE
chore: move sales-order.py code to erpnext from bloomstack_core

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
https://github.com/Bloomstack/bloomstack_core/pull/689
screencast-bloomstack_demo_8000-2021 09 14-12_56_59

### CODE REVIEW

**Review by vishdha — CHANGES_REQUESTED** [review]
update requested change in bloomstac_core Pr then it's good for merging

**Review by vishdha — APPROVED** [review]

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 1]
```diff
@@ -64,6 +66,22 @@ def validate(self):
...
+			self.docstatus == 1,
+			self.status not in ["On Hold", "Closed", "Completed"],
+			self.skip_delivery_note == 0,
+			flt(self.per_delivered, 6) < 100,
+			getdate(self.delivery_date) < getdate(today()),
+		]
+
+		is_overdue = all(overdue_conditions)
+		if is_overdue != self.is_overdue:
+			self.db_set("is_overdue", is_overdue)
```
Comment (sahil28297) [inline erpnext/selling/doctype/sales_order/sales_order.py:83]: Check if `db_set` is to be used here, it triggers before_change and on_change events, also it does not commit.

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
Attach a gif that contains the execution of all the functions that are moved in this PR.

File: erpnext/selling/doctype/sales_order/sales_order.py (hunk 1 above)
Comment (sahil28297) [reply erpnext/selling/doctype/sales_order/sales_order.py:83]: @MyuddinKhatri did you test this?

Comment (MyuddinKhatri) [reply erpnext/selling/doctype/sales_order/sales_order.py:83]: Yes i did test this it is working fine

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**